src/elements/toniq-pagination/toniq-pagination.element.ts

```diff
@@ -0,0 +1,220 @@
+import {assign, css, defineElementEvent, html, listen} from 'element-vir';
+import {classMap} from 'lit/directives/class-map.js';
+import {map} from 'lit/directives/map.js';
+import {clamp} from '../../augments/number';
+import {ArrowLeft24Icon, ArrowRight24Icon} from '../../icons';
+import {
+    applyBackgroundAndForeground,
+    removeNativeButtonStyles,
+    toniqColors,
+    toniqFontStyles,
+} from '../../styles';
+import {defineToniqElement} from '../define-toniq-element';
+import {ToniqButton, ToniqButtonVariant} from '../toniq-button/toniq-button.element';
+import {ToniqIcon} from '../toniq-icon/toniq-icon.element';
+
+export const ToniqPagination = defineToniqElement({
+    tagName: 'toniq-pagination',
+    props: {
+        currentPage: 1,
+        /** This is required to set the total pages */
+        pageCount: 10,
+        pageSize: 7,
+    },
+    styles: css`
+        :host {
+            display: flex;
+            align-items: center;
+            justify-content: center;
+        }
+
+        ${ToniqButton}::part(inner-button) {
+            margin: 0 8px;
+            padding: 4px;
+        }
+
+        ${ToniqButton}::part(inner-button):hover {
+            ${applyBackgroundAndForeground(toniqColors.accentPrimary)};
+        }
+
+        ${ToniqButton}.disabled::part(inner-button) {
+            pointer-events: none;
+            ${applyBackgroundAndForeground(toniqColors.accentTertiary)};
+        }
+
+        button {
+            ${removeNativeButtonStyles}
+        }
+
+        .page {
+            position: relative;
+            width: 32px;
+            height: 32px;
+            ${toniqFontStyles.labelFont}
+        }
+
+        .page[disabled],
+        .page.selected {
+            cursor: auto;
+        }
+
+        .page:not(.selected):not([disabled]):hover {
+            ${applyBackgroundAndForeground(toniqColors.pageInteraction)};
+        }
+
+        .page.selected {
+            color: white;
+        }
+
+        .page.selected::after {
```

Comment:
Why an `::after` pseudo selector?

Reply:
The after is used to add the background for current page
![Screen Shot 2022-07-01 at 12 27 48 PM](https://user-images.githubusercontent.com/99699764/176823438-9cce3087-2b25-4868-81f4-0805f7f8e81e.png)

Reply:
Don't use `::after` for this. This should be handled with the `button` vs `toniq-button` discussion.